### DISCUSSION TITLE
Is there a plan to verify the SecretKey complexity on the Minio server?

### DISCUSSION THREAD

**Discussion by LiYvbo** [minio/minio#11439] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
When i read the source code, I found minIO only verifies the length of secretKey.  I wonder if there is a plan to make a more strict verfication for the AccessKey/SecretKey? 

**Describe the solution you'd like**
SecretKey check list, such as:
- sepecial characters
- numeric characters
- alphabet...

**Describe alternatives you've considered**
The current check method is too weak.

**Additional context**
![image](https://user-images.githubusercontent.com/9931878/106865564-53e65200-6706-11eb-8d81-4235dab3cbae.png)

**Comment by harshavardhana** [comment, marked as answer]
> When i read the source code, I found minIO only verifies the length of secretKey. I wonder if there is a plan to make a more strict verfication for the AccessKey/SecretKey?

@LiYvbo we have many users with many different requirements, the only thing we can validate sanely to quality all the requirements is the length. 

It is just a best practice that we expect that users set them to good values. I don't think we can enforce anything meaningful other than character length at this point in time.
